src/geo/transform.js

```diff
@@ -1600,6 +1623,7 @@ class Transform {
         this.pixelMatrix = mat4.multiply(new Float64Array(16), this.labelPlaneMatrix, this.projMatrix);
 
         this._calcFogMatrices();
+        this._calcDistanceMatrices();
```

Comment:
Not something to address now, but as a potential follow up to this PR: Is there any higher level hint we could easily use for clipping (e.g. the existence of _any_ clip expression in the style) to determine whether these matrices should be calculated?

I imagine most styles wouldn't use it immediately as it gets introduced, so this may reduce the impact to the most common code path. Not that it's a lot of extra operations, but I'm realizing that this also applies to fog, where we could skip `this._calcFogMatrices` if fog isn't enabled.